install/0000_80_machine-config-operator_07_etcdquorumguard_deployment.yaml

```diff
@@ -49,16 +50,21 @@ spec:
         operator: Exists
         effect: NoSchedule
       containers:
-      - image: registry.svc.ci.openshift.org/openshift:cli
+      - image: quay.io/openshift/origin-cli:latest
         imagePullPolicy: IfNotPresent
         name: guard
         volumeMounts:
         - mountPath: /mnt/kube
           name: kubecerts
         command:
-        - "/bin/sleep"
+        - /bin/bash
         args:
-        - "infinity"
+        - -c
+        - |
+          # properly handle TERM and exit as soon as it is signaled
```

Comment:
We don't need to dive deep into this now, but...man this code.  Why isn't `sleep` just dying on `SIGTERM`?  I think this *might* be related to something in `runc` setting `SIGTERM` to `SIG_IGN` and passing that onto its children; not totally sure yet.

Reply:
The issue that Clayton noted is that pid1 -- and the command run by the container has pid #1 -- is treated specially by Linux; it ignores all signals unless the process specifically registers a handler for the signal.  IIRC this is to prevent `init` (which is `systemd` these days) from being killed and immediately rebooting the system, as older UNIX systems did.

I think we could simply make the command something like

```
/bin/sh -c 'trap exit TERM; sleep infinity'
```

without all the other stuff.